### PR TITLE
[WIP] Docs/events

### PR DESCRIPTION
**Pull request by afflom**
This PR introduces Events to the UOR Framework.

### CODE REVIEW

**Comment by afflom** [comment]
@d10n @jpower432 can you please let me know which parts of this doc need clarification?

**Review by jpower432 — COMMENTED** [review]
A couple questions:

File: docs/design/events.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+### Router API
+| Control Message   |
+|-------------------|
+| attributes (map)  | The attributes of the object
+| digest            | The digest of the object
+| source            | The source address of the object
+| destination       | The destination address of the object
+| payload (bool)    | If no payload, the object is only attributes
+
+# Event Engine Design
```
[the remark this answers is not in the document]
Comment (sabre1041) [reply docs/design/events.md:35]: +1

File: docs/design/events.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Event Engines are designed as follows:
+1. Event Engines can be UOR native applications or legacy applications ported to UOR.
```
Comment (sabre1041) [inline docs/design/events.md:38]: Do we have definitions for "UOR native" applications?

File: docs/design/events.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+2. An application becomes an Event Engine when it implements the UOR Router API and performs tasks pertinent to a UOR Collection.
```
Comment (sabre1041) [inline docs/design/events.md:39]: Is the Router an Interface? I see a Control Message, but what else is there defined aside from the 2 routing actions and the Control Message

Also, what are tasks "pertinent" to a UOR collection?

File: docs/design/events.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+# Event Engine Design
+
+Event Engines are designed as follows:
+1. Event Engines can be UOR native applications or legacy applications ported to UOR.
+2. An application becomes an Event Engine when it implements the UOR Router API and performs tasks pertinent to a UOR Collection. 
+3. Event Engines receive application control signalling encoded in attributes. 
+4. UOR Native Applications implement the UOR local cache structure, while UOR ported applications rely on URL encoded attribute maps to reference resources.
+
+Event Engine's follow the following workflow:
```
Comment (sabre1041) [inline docs/design/events.md:43]: Use a different work instead of "follow the following"

**Pull request closed without merging**